src/inmanta/env.py

```diff
@@ -424,94 +390,118 @@ def compose_list_command(
-        allow_pre_releases: bool = False,
         constraints_files: Optional[List[str]] = None,
-        requirements_files: Optional[List[str]] = None,
-        use_pip_config: Optional[bool] = False,
+        paths: Optional[List[LocalPackagePath]] = None,
     ) -> None:
-        cmd: List[str] = PipCommandBuilder.compose_install_command(
-            python_path=python_path,
-            requirements=requirements,
-            paths=paths,
-            index_urls=index_urls,
-            upgrade=upgrade,
-            upgrade_strategy=upgrade_strategy,
-            allow_pre_releases=allow_pre_releases,
-            constraints_files=constraints_files,
-            requirements_files=requirements_files,
-            use_pip_config=use_pip_config,
+        """
+        Perform a pip installaccording to the given config
+
+        :param python_path: the python path to use
+        :param config: the pip config to use
+
+        :param requirements: which requirements to install
+        :param requirements_files: which requirements_files to install (-r)
+        :param paths: which paths to install
+
+        :param constraints_files: pass along the following constraint files
+
+        :param upgrade: make pip do an upgrade
+        :param upgrade_strategy: what upgrade strategy to use
+        """
+
+        # What
+        requirements = requirements if requirements is not None else []
+        requirements_files = requirements_files if requirements_files is not None else []
+
+        paths = paths if paths is not None else []
+        local_paths: Iterator[LocalPackagePath] = (
+            # make sure we only try to install from a local source: add leading `./` and trailing `/` to explicitly tell pip
+            # we're pointing to a local directory.
+            LocalPackagePath(path=os.path.join(".", path.path, ""), editable=path.editable)
+            for path in paths
         )
+
+        install_args = [
+            *(str(requirement) for requirement in requirements),
+            *chain.from_iterable(["-r", f] for f in requirements_files),
+            *chain.from_iterable(["-e", path.path] if path.editable else [path.path] for path in local_paths),
+        ]
+
+        # From where
+        if paths:
+            # For local installs, we allow not having an index set.
+            pass
+        else:
+            # All others need an index
+            config.assert_has_source(" ".join(install_args))
+
+        index_args: list[str] = []
+        if config.index_url:
+            index_args.append("--index-url")
+            index_args.append(config.index_url)
+        elif not config.use_system_config:
+            # If the config doesn't set index url
+            # and we are not using system config,
+            # then we need to disable the index.
+            # This can only happen if paths is also set.
+            index_args.append("--no-index-url")
```

Comment:
I think this is the option:
```suggestion
            index_args.append("--no-index")
```